syft/pkg/cataloger/binary/cataloger_test.go

```diff
@@ -69,12 +70,26 @@ func Test_Cataloger_DefaultClassifiers_PositiveCases(t *testing.T) {
 				Version:   "9.5alpha1",
 				Type:      "binary",
 				PURL:      "pkg:generic/postgresql@9.5alpha1",
-				Locations: singleLocation("postgres"),
+				Locations: locations("postgres"),
 				Metadata: pkg.BinaryMetadata{
 					Classifier: "postgresql-binary",
 				},
 			},
 		},
+		{
+			name:       "positive-python-duplicates",
+			fixtureDir: "test-fixtures/classifiers/positive/python-duplicates",
+			expected: pkg.Package{
+				Name:      "python",
+				Version:   "3.8.16",
+				Type:      "binary",
+				PURL:      "pkg:generic/python@3.8.16",
+				Locations: locations("dir/python3.8", "python3.8", "libpython3.8.so", "patchlevel.h"),
+				Metadata: pkg.BinaryMetadata{
+					Classifier: "python-binary+python-binary-lib+cpython-source",
```

Comment:
If we're going to merge multiple findings together we probably shouldn't combine the names like this. Maybe change to `Classifier []string`, or `Metadata` stores `[]pkg.BinaryMetadata`, or `pkg.BinaryMetadata` has a collection within it that encompasses more fields.

Reply:
After offline discussion how to best represent this, I've updated things to use this structure:

```go
type BinaryMetadata struct {
	Matches []ClassifierMatch `mapstructure:"Matches" json:"matches"`
}

type ClassifierMatch struct {
	Classifier string          `mapstructure:"Classifier" json:"classifier"`
	Location   source.Location `mapstructure:"Location" json:"location"`
}
```